### PR TITLE
Prune render

### PR DESCRIPTION
**Pull request by anupriyatripathi**
This PR allows users to prune qemistrees using any column in feature data. 
- It removes the stringency that pruning needs to be done using classyfire columns only. 
- If column is not specified, it filters out all the feature not present in feature data.
- Labels tips by parent mz (if specified), when feature in unclassified

In the future, it will allow us to build on this and integrate other feature metadata fields that could be used to prune and visualize qemistrees (such as parent mass)

### CODE REVIEW

File: q2_qemistree/_prune_hierarchy.py [hunk 1]
```diff
@@ -11,8 +11,8 @@
 
 
 def prune_hierarchy(feature_data: pd.DataFrame, tree: TreeNode,
-                    column: str) -> TreeNode:
-    '''Prunes the tips of the tree to remove unannotated entries
+                    column: str = None) -> TreeNode:
+    '''Prunes the tips of the tree to keep feature of interest
```
Comment (ElDeveloper) [inline q2_qemistree/_prune_hierarchy.py]:
```suggestion
    '''Prunes the tips of the tree to keep features of interest
```

File: q2_qemistree/_prune_hierarchy.py [hunk 2]
```diff
@@ -39,27 +37,15 @@ def prune_hierarchy(feature_data: pd.DataFrame, tree: TreeNode,
...
-                                                       ', '.join(known)))
-
-    failed_values = {'unclassified', 'unexpected server response',
-                     'SMILE parse error'}
-
-    # remove all NA values or missing values
-    feature_data = feature_data[~(feature_data[column].isin(failed_values) |
-                                  feature_data[column].isna())]
-
+    if column:
```
Comment (ElDeveloper) [inline q2_qemistree/_prune_hierarchy.py]: This line isn't needed, since you are already checking this in line 40.

Comment (anupriyatripathi) [reply q2_qemistree/_prune_hierarchy.py]: 👍 you're so right

Comment (anupriyatripathi) [reply q2_qemistree/_prune_hierarchy.py]: actually, I think we need it. See changelogs

Comment (ElDeveloper) [reply q2_qemistree/_prune_hierarchy.py]: Oh, I see! Can you add a note in plugin setup that describes this behavior? The best place would be under "column" description i.e. `Features with missing values in this column will be removed from the tree. If no column name is specified then the tree will be pruned to only contain tips in the feature data.`

**Comment by ElDeveloper** [comment]
Thanks @anupriyatripathi

**Pull request merged**